internal/storage/bucket_handle_test.go

```diff
@@ -467,3 +469,67 @@ func (t *BucketHandleTest) TestUpdateObjectMethodWithMissingObject() {
 
 	AssertTrue(errors.As(err, &notfound))
 }
+
+func (t *BucketHandleTest) TestComposeObjectMethodWithValidObject() {
+	obj, err := t.bucketHandle.StatObject(context.Background(),
+		&gcs.StatObjectRequest{
+			Name: TestObjectName,
+		})
+
+	AssertEq(nil, err)
+
+	composedObj, err := t.bucketHandle.ComposeObjects(context.Background(),
+		&gcs.ComposeObjectsRequest{
+			DstName:                       TestObjectName,
+			DstGenerationPrecondition:     nil,
+			DstMetaGenerationPrecondition: nil,
+			Sources: []gcs.ComposeSource{
+				{
+					Name: TestSubObjectName,
+				},
+			},
+			ContentType:     ContentType,
+			ContentEncoding: ContentEncoding,
+			ContentLanguage: ContentLanguage,
+			CacheControl:    CacheControl,
+			CustomTime:      CustomTime,
+			EventBasedHold:  true,
+			StorageClass:    StorageClass,
+			Metadata: map[string]string{
+				MetaDataKey: MetaDataValue,
+			},
+			Acl: nil,
+		})
+
+	AssertEq(nil, err)
+	AssertNe(nil, obj)
+	AssertGe(composedObj.Size, obj.Size)
+}
+
+func (t *BucketHandleTest) TestComposeObjectMethodWithInValidObject() {
```

Comment:
can you add a test case to compose from two source objects

Reply:
fixed